hack/operatorhub/main.go

```diff
@@ -615,19 +626,23 @@ func getImageDigest(apiKey, projectId, version string) (string, error) {
 	if err := json.NewDecoder(res.Body).Decode(&images); err != nil {
 		return "", err
 	}
-	if len(images.Data) > 1 {
-		fmt.Fprintf(os.Stderr, "\nid                       creation_date                    docker_image_digest\n")
-		for _, image := range images.Data {
-			fmt.Fprintf(os.Stderr, "%s %s %s\n", image.Id, image.CreationDate, image.DockerImageDigest)
-		}
-		return "", fmt.Errorf("found %d images with tag %s in RedHat catalog while only one is expected", len(images.Data), version)
-	}
+
 	if len(images.Data) == 0 {
-		return "", fmt.Errorf("no image with tag %s in RedHat catalog", version)
+		return "", fmt.Errorf("no published, undeleted images found in RedHat catalog")
+	}
+
+	var imageDigest string
+	for _, image := range images.Data {
+		for _, repo := range image.Repositories {
+			for _, tag := range repo.Tags {
+				if tag.Name == version {
+					imageDigest = image.DockerImageDigest
+				}
+			}
+		}
 	}
-	imageDigest := images.Data[0].DockerImageDigest
 	if imageDigest == "" {
-		return "", fmt.Errorf("image digest for %s is empty", version)
+		return "", fmt.Errorf("no recent images found for %s", version)
```

Comment:
nit: It might be helpful to let the user know that only 4 images have been retrieved from the API.

```suggestion
		return "", fmt.Errorf("no published and undeleted image for version %s found in the %d most recent items returned by the catalog API", version, MAX_IMAGES)
```